integration/hsic/hsic.go

```diff
@@ -203,6 +216,27 @@ func WithEmbeddedDERPServerOnly() Option {
 	}
 }
 
+// WithCustomDERPServerOnly configures Headscale use a custom
+// DERP server only.
+func WithCustomDERPServerOnly(contents []byte) Option {
```

Comment:
Maybe this could be named `WithDERPConfig`, since it is more general dumping a file and using that?